### ISSUE TITLE
Fitting parameters to other datasets

### ISSUE THREAD

**Issue by AdeeshKolluru** [grimme-lab/dxtb#164]
Hey, great work on putting this together! 
I was curious if learning and fitting the parameters (of maybe GFN1 or so) on DFT datasets is possible. If so, do you have an example of that?

**Comment by njzjz** [comment]
> Yes, learning and fitting GFN1 parameters on DFT datasets is possible.

Hi, which parameters can be fit? I don't find `torch.nn.Parameter` being used.

**Comment by marvinfriede** [comment]
Hey @brainiac-mxn,

> Is this type of data (structures + energies + gradients) suitable for the reparametrization process within dxtb?

Yes, this data should work. I don't see any reason why not. Just remember that, if your reference energies are DFT energies, they are on a different scale than the xTB energies. So, you either need relative energies or some sort of atomic increment (as used in g-xTB).
 
> I have been unable to find a dedicated tutorial for this specific task. Could anyone point me to relevant documentation or examples?

Unfortunately, we have no ready-to-use public setup yet, as this is still a work in progress. I can only point you to the [example](https://github.com/grimme-lab/dxtb/blob/main/examples/optim/param.py) showing how to calculate derivatives of the parameters. The main entry point is the `dxtb.ParamModule`, which basically turns the whole (pydantic) parametrization object into a `torch.nn.Module`, which allows building a PyTorch training pipeline.
